templates/a_muon.cpp.in

```diff
@@ -102,19 +318,30 @@ double calculate_a_muon_impl(const @ModelName@_mass_eigenstates& model, const so
       return std::numeric_limits<double>::quiet_NaN();
    }
 
-   double val = 
+   double val =
       // vector form factor
       1./2.*(form_factors[2] + form_factors[3]).real()
-      // reinstate the mass that was factored out in definition of form factor 
-      * context.mass<Muon>({@AMuon_MuonIndex@}) 
+      // reinstate the mass that was factored out in definition of form factor
+      * context.mass<Muon>({@AMuon_MuonIndex@})
       // factor out e/(2*muon pole mass)
-      / (unit_charge(context_base{model})/(2.*muonPhysicalMass(qedqcd))) 
+      / (unit_charge(context_base{model})/(2.*muonPhysicalMass(qedqcd)))
       // definition of photon momentum flow for g-2 is opposite than for form factors
       * (-1.);
 
    // add 2-loop QED logarithms
    val *= 1. + get_QED_2L(context, qedqcd);
 
+   std::complex<double> valBarZee {0., 0.};
+
+@AMuon_BarZeeCalculation@
+
+   if (!is_zero(valBarZee.imag())) {
+      ERROR("Error in the g-2 calculation! Form factor F2 should be real");
+      return std::numeric_limits<double>::quiet_NaN();
+   } else {
+      val += valBarZee.real();
```

Comment:
If we want to keep the Barr-Zee diagrams, but not activate then, we may simply comment out this line.

Reply:
Done in b8e7808.